File: probs/1931.py
import sys
input = sys.stdin.readline

N = int(input())
meetings = []
for _ in range(N):
    meetings.append(tuple(map(int, input().split())))
meetings.sort(key=lambda x: x[0])
meetings.sort(key=lambda x: x[1])

fin, cnt = 0, 0
for meeting in meetings:
    if meeting[0] >= fin:
        fin = meeting[1]
        cnt += 1
print(cnt)

File: probs/1931.cpp
#include <bits/stdc++.h>

using namespace std;
using pi = pair<int, int>;

const int MAX = 1e5;

pi A[MAX];

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    freopen("input.txt", "r", stdin);

    int N; cin >> N;
    for (int i = 0; i < N; i++) {
        int s, e; cin >> s >> e;
        A[i] = {e, s};
    }
    sort(A, A+N);

    int t = 0, cnt = 0;
    for (int i = 0; i < N; i++) {
        auto [e, s] = A[i];
        if (s >= t) {
            t = e;
            cnt++;
        }
    }

    cout << cnt << '\n';

    return 0;
}
